lib/kitchen/note_element.rb

```diff
@@ -62,10 +58,26 @@ def self.is_the_element_class_for?(node)
 
     protected
 
-    def translation_key_in(possible_translation_keys)
-      keys = possible_translation_keys & classes
+    def key
+      if I18n.t(document.short_name).instance_of?(String)
+        puts "Entry for book with short name #{document.short_name} does not exist in locales"
+        return
+      end
+      unless I18n.t(document.short_name).key?(:notes)
+        puts "The book #{document.short_name} does not have an entry for notes in locales"
+        return
+      end
+
+      possible_keys = I18n.t(document.short_name)[:notes].keys.map(&:to_s)
+      keys = possible_keys & classes
+
       raise("too many translation keys: #{keys.join(', ')}") if keys.many?
 
+      unless keys.first.present?
+        puts "Title for note with classes #{classes} not found"
+        return
+      end
+
       keys.first
```

Comment:
Memoize this whole method -- we call it twice, once in `indicates_autogenerated_title?` and then again when we do the actual translation, no need to run all the code twice.

```ruby
def detected_note_title_key
  @detected_note_title_key ||= begin
    ... all the code ending with `keys.first` which is the return value for this block and will store in the var
  end
end
```